src/block/Button.php

```diff
@@ -91,4 +89,19 @@ public function onNearbyBlockChange() : void{
 	private function canBeSupportedAt(Block $block, int $face) : bool{
 		return $block->getAdjacentSupportType(Facing::opposite($face))->hasCenterSupport();
 	}
+
+	public function onProjectileInteraction(Projectile $projectile) : void{
+		if($projectile instanceof WindCharge) {
+			$this->press();
+		}
+	}
+
+	public function press() : void {
```

Comment:
This should be `private` or `protected`, making it public has implications beyond this PR's scope